app/services/geolocalization_services/user_localization_and_distance.py

```diff
@@ -0,0 +1,83 @@
+import random
+import country_converter as coco
+import requests
+import json
+from ip2geotools.databases.noncommercial import DbIpCity
+from geopy.distance import geodesic
+from geopy.geocoders import Nominatim
+
+
+def get_coordinates(city_name):  # Generates latitude and longtude from city name
+    geolocator = Nominatim(user_agent="Botinder")
+    location = geolocator.geocode(city_name)
+
+    if location:
+        latitude = location.latitude
+        longitude = location.longitude
+        return latitude, longitude
+    else:
+        return None, None
+
+
+def get_cities():
+    location = get_location()
+    country_code = location["country"]
+    with open("app/geolocalization_services/cities.json", "r") as cities_data:

```

Comment:
Loading the file every single time when you use the function is not the optimal solution 😃